src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -137,9 +131,9 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
                 $operation['path'] .= sprintf('/%s%s', $this->pathSegmentNameGenerator->getSegmentName($property, $operation['collection']), self::FORMAT_SUFFIX);
             }
 
-            $tree[$visiting] = $operation;
+            $tree[$operation['route_name']] = $operation;
 
-            $this->computeSubresourceOperations($subresourceClass, $tree, $rootResourceClass, $operation, $visiting);
+            $this->computeSubresourceOperations($subresourceClass, $tree, $rootResourceClass, $operation, array_merge($visited, [$visiting]));
```

Comment:
The result would be wrong because we have a loop, if we modify the `$visited` array in place it'd affect the other properties.